apps/dashboard/app/models/remote_file.rb

```diff
@@ -0,0 +1,89 @@
+require "rclone_util"
+
+class RemoteFile
+  attr_reader :path, :full_path, :remote
+
+  delegate :basename, :descend, :parent, :join, :to_s, to: :full_path
+
+  def initialize(path, remote)
+    # accepts both String and Pathname
+    # avoids converting to Pathname in every function
+    @path = Pathname.new(path)
+    @remote = remote
+    @full_path = Pathname.new("#{@remote}:#{@path}")
+  end
+
+  def remote_type
+    @remote_type ||= RcloneUtil.remote_type(remote)
+  end
+
+  def raise_if_cant_access_directory_contents
+  end
+
+  def directory?
+    RcloneUtil.directory?(@remote, @path)
+  end
+
+  def ls
+    files = RcloneUtil.ls(@remote, path)
+    # Need to return same fields as PosixFile.
+    # owner, mode and dev are not defined for remote files, leaving them empty
+    files.map do |file|
+      {
+        id: file["Path"],
+        name: file["Name"],
+        size: file["IsDir"] ? nil : file["Size"],
+        human_size: file["IsDir"] ? "-" : ::ApplicationController.helpers.number_to_human_size(file["Size"], precision: 3),
+        directory: file["IsDir"],
+        date: DateTime.parse(file["ModTime"]).to_time.to_i,
+        owner: "",
+        mode: "",
+        dev: 0,
+      }
+    end.sort_by { |p| p[:directory] ? 0 : 1 }
+  end
+
+  def can_download_as_zip?(timeout: Configuration.file_download_dir_timeout, download_directory_size_limit: Configuration.file_download_dir_max)
+    [false, "Downloading remote files as zip is currently not supported"]
+  end
+
+  def editable?
+    true
+  end
+
+  def read(&block)
+    RcloneUtil.cat(@remote, @path, &block)
+  end
+
+  def touch
+    RcloneUtil.touch(@remote, @path)
+  end
+
+  def mkdir
+    RcloneUtil.mkdir(@remote, @path)
+  end
+
+  def write(content)
+    RcloneUtil.write(@remote, @path, content)
+  end
+
+  def handle_upload(tempfile)
+    # This is not optimal as the file is first uploaded fully to the OOD server
+    # and then transfered to the remote
```

Comment:
```suggestion
    # FIXME: upload to the remote asynchronously 
```

Reply:
Fixed

Reply:
Cool. It may be nit-picky but IDEs and even rails can search these files for FIXME and TODO. `bin/rails notes` will do the same.